File: app/src/main/java/hr/ml/moviesdiscover/viewmodel/SharedDataViewModel.java
package hr.ml.moviesdiscover.viewmodel;

import android.util.Log;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import hr.ml.moviesdiscover.model.Movie;

public class SharedDataViewModel extends ViewModel {
    private static final String TAG = "SharedDataViewModel";

    public MutableLiveData<Movie> selectedMovie = new MutableLiveData<>();

    public SharedDataViewModel() {
        Log.d(TAG, "SharedDataViewModel: created");
    }

    public void setSelectedMovieId(Movie movie) {
        selectedMovie.setValue(movie);
    }

}

File: app/src/main/kotlin/hr/ml/moviesdiscover/viewmodel/SharedDataViewModel.kt
package hr.ml.moviesdiscover.viewmodel

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import hr.ml.moviesdiscover.model.Movie

private const val TAG = "SharedViewModel"

class SharedDataViewModel : ViewModel() {

    val selectedMovie = MutableLiveData<Movie>()

    init {
        Log.d(TAG, "created")
    }

}